x-pack/plugins/lens/public/indexpattern_datasource/state_helpers.ts

```diff
@@ -4,162 +4,7 @@
-  columnId: string
-) {
-  const newColumns = { ...columns };
-  Object.keys(newColumns).forEach((currentColumnId) => {
-    if (currentColumnId !== columnId) {
-      const currentColumn = newColumns[currentColumnId];
-      const operationDefinition = operationDefinitionMap[currentColumn.operationType];
-      newColumns[currentColumnId] = operationDefinition.onOtherColumnChanged
-        ? operationDefinition.onOtherColumnChanged(currentColumn, newColumns)
-        : currentColumn;
-    }
-  });
-  return newColumns;
-}
-
-export function changeColumn<C extends IndexPatternColumn>({
-  state,
-  layerId,
-  columnId,
-  newColumn,
-  keepParams = true,
-}: {
-  state: IndexPatternPrivateState;
-  layerId: string;
-  columnId: string;
-  newColumn: C;
-  keepParams?: boolean;
-}): IndexPatternPrivateState {
-  const oldColumn = state.layers[layerId].columns[columnId];
-
-  const updatedColumn =
-    keepParams &&
-    oldColumn &&
-    oldColumn.operationType === newColumn.operationType &&
-    'params' in oldColumn
-      ? { ...newColumn, params: oldColumn.params }
-      : newColumn;
-
-  if (oldColumn && oldColumn.customLabel) {
-    updatedColumn.customLabel = true;
-    updatedColumn.label = oldColumn.label;
-  }
-
-  const layer = {
-    ...state.layers[layerId],
-  };
-
-  const newColumns = adjustColumnReferencesForChangedColumn(
-    {
-      ...layer.columns,
-      [columnId]: updatedColumn,
-    },
-    columnId
-  );
-
-  return mergeLayer({
-    state,
-    layerId,
-    newLayer: {
-      columnOrder: getColumnOrder({
-        ...layer,
-        columns: newColumns,
-      }),
-      columns: newColumns,
-    },
-  });
-}
-
-export function deleteColumn({
-  state,
-  layerId,
-  columnId,
-}: {
-  state: IndexPatternPrivateState;
-  layerId: string;
-  columnId: string;
-}): IndexPatternPrivateState {
-  const hypotheticalColumns = { ...state.layers[layerId].columns };
-  delete hypotheticalColumns[columnId];
-
-  const newColumns = adjustColumnReferencesForChangedColumn(hypotheticalColumns, columnId);
-  const layer = {
-    ...state.layers[layerId],
-    columns: newColumns,
-  };
-
-  return mergeLayer({
-    state,
-    layerId,
-    newLayer: {
-      ...layer,
-      columnOrder: getColumnOrder(layer),
-    },
-  });
-}
-
-export function getColumnOrder(layer: IndexPatternLayer): string[] {
-  const [aggregations, metrics] = _.partition(
-    Object.entries(layer.columns),
-    ([id, col]) => col.isBucketed
-  );
-
-  return aggregations
-    .sort(([id, col], [id2, col2]) => {
-      return (
-        // Sort undefined orders last
-        (col.suggestedPriority !== undefined ? col.suggestedPriority : Number.MAX_SAFE_INTEGER) -
-        (col2.suggestedPriority !== undefined ? col2.suggestedPriority : Number.MAX_SAFE_INTEGER)
-      );
-    })
-    .map(([id]) => id)
-    .concat(metrics.map(([id]) => id));
-}
+import { IndexPatternPrivateState, IndexPatternLayer } from './types';
 
 export function mergeLayer({
```

Comment:
This `mergeLayer` function is kept in the `state_helpers` code, every other function has moved because it's dependent on the operation definitions.